WCFDataService/Client/System/Data/Services/Client/Materialization/ODataEntityMaterializer.cs

```diff
@@ -970,11 +970,9 @@ private void MergeLists(MaterializerEntry entry, ClientPropertyAnnotation proper
             Debug.Assert(property != null, "property != null");
             Debug.Assert(plan != null || nextLink == null, "plan != null || nextLink == null");
 
-            // Simple case: the list is of the target type, and the resolved entity
-            // has null; we can simply assign the collection. No merge required.
+            // Simple case: the list is of the target type; we can simply assign the collection. No merge required.
             if (entry.ShouldUpdateFromPayload &&
-                property.NullablePropertyType == list.GetType() &&
-                property.GetValue(entry.ResolvedObject) == null)
```

Comment:
Hi @Zoe-ms 
I am now confident that this is the correct and simple solution.
I have reverted all prior commits and implemented the simple fix from scratch.
MergeLists() is only called during projection from RebindEntityMemberInit, so we are only called with projected queries having a select part in the form "select new File { FileRecords = ...,, }", where FileRecords is a Collection. It is actually rather wrong that this could lead to anything else than an assignment to the FileRecords property, but it always did as FileRecords is never null in a standard proxy.
I have an additional test scenario for you to validate (my prior attempt did not fix this, but this one does)...
Say we have a server that uses paging with a page size of 50 and we have a File with 52 FileRecords on it, then running the following test code...

`            var ctx = Context();
            ctx.Format.UseJson();

            var q = from f in ctx.Files
                    where f.ID == "2968889"
                    select new File
                    {
                        ID = f.ID,
                        Title = f.Title,
                        Officer_Value = f.Officer_Value,
                        ResponsibleOu_Value = f.ResponsibleOu_Value,
                        FileRecords = f.FileRecords // FileRecords is a DSC<> type. f.FileRecords has TrackingMode.None as expected - we do not get change tracking of FileRecords
                    };

            var file = q.Single();

            //Paging thru the FileRecords DataServiceCollection property.
            while (file.FileRecords.Continuation != null)
                file.FileRecords.Load(ctx.Execute(file.FileRecords.Continuation));

            Console.WriteLine(file.FileRecords.Count); // Outputs 52

            foreach (Record rec in file.FileRecords)
            {
                rec.Officer_Value = file.Officer_Value;
                Console.WriteLine(ctx.GetEntityDescriptor(rec).State); // Outputs 52 * Unchaged as expected.
            }

            // Now lets use MergeOption.OverwriteChanges and GET the same File, but this time we assign a TrackingMode.AutoChangeTracking DSC<> to FileRecords 
            ctx.MergeOption = MergeOption.OverwriteChanges;
            q = from f in ctx.Files
                    where f.ID == "2968889"
                    select new File
                    {
                        ID = f.ID,
                        Title = f.Title,
                        Officer_Value = f.Officer_Value,
                        ResponsibleOu_Value = f.ResponsibleOu_Value,
                        FileRecords = new DataServiceCollection<Record> // This constructor sets TrackingMode to TrackingMode.AutoChangeTracking
                           (
                               from r in f.FileRecords
                               select new Record
                               {
                                   ID = r.ID,
                                   AccessCode = r.AccessCode
                               }
                           )
                    };

            file = q.Single();
            while (file.FileRecords.Continuation != null)
                file.FileRecords.Load(ctx.Execute(file.FileRecords.Continuation));

            Console.WriteLine(file.FileRecords.Count); // Outputs 52

            foreach (Record rec in file.FileRecords)
            {
                rec.Officer_Value = file.Officer_Value;
                Console.WriteLine(ctx.GetEntityDescriptor(rec).State); // Should output 52 * Modified, but did only output 50 * Modified and the 2 * Unchanged
            }
`